src/Core/src/Platform/Windows/ButtonExtensions.cs

```diff
@@ -159,10 +160,18 @@ public static void UpdateImageSource(this Button platformButton, WImageSource? n
 		{
 			if (platformButton.GetContent<WImage>() is WImage nativeImage)
 			{
-				// Stretch to the size of the button
+				// Stretch to fill
 				nativeImage.Stretch = UI.Xaml.Media.Stretch.Uniform;
 				nativeImage.Source = nativeImageSource;
 
+				// If we're a CanvasImageSource (font image source), we need to explicitly set the image height
+				// to the desired size of the font, otherwise it will be stretched to the available space
+				if (nativeImage.Source is CanvasImageSource canvas)
+				{
+					nativeImage.Width = canvas.Size.Width;
+					nativeImage.Height = canvas.Size.Height;
+				}
```

Comment:
I feel like I reviewed a PR weeks ago that removed all the measuring code here - from normal images and things.

Reply:
This commit: https://github.com/dotnet/maui/commit/7dff6eb55b85ae50bc9e0ce90778981b913e34e1#diff-e34feff8430d43e27f52abff099ba347b41628f6bde6a68e12f8b618722c59c9

Should this rather have been implemented differently instead of just being removed?

Reply:
Yeah, you did 🥲. I misunderstood the behavior of how `FontImageSource` scales-to-fit the available space, which differs from how images in buttons scale-to-fit.